dbfit-java/informix/src/main/java/dbfit/environment/InformixEnvironment.java

```diff
@@ -0,0 +1,341 @@
+        qry += "                                               WHEN 14 THEN 'FRACTION(4)'";
+        qry += "                                               WHEN 15 THEN 'FRACTION(5)'";
+        qry += "                                           END";
+        qry += "            WHEN   11 THEN 'BYTE'";
+        qry += "            WHEN   12 THEN 'TEXT'";
+        qry += "            WHEN   13 THEN 'VARCHAR'";
+        qry += "            WHEN   14 THEN 'INTERVAL'";
+        qry += "            WHEN   15 THEN 'NCHAR'";
+        qry += "            WHEN   16 THEN 'NVARCHAR'";
+        qry += "            WHEN   17 THEN 'INT8'";
+        qry += "            WHEN   18 THEN 'SERIAL8'";
+        qry += "            WHEN   19 THEN 'SET'";
+        qry += "            WHEN   20 THEN 'MULTISET'";
+        qry += "            WHEN   21 THEN 'LIST'";
+        qry += "            WHEN   22 THEN 'ROW'";
+        qry += "            WHEN   23 THEN 'COLLECTION'";
+        qry += "            WHEN   24 THEN 'ROWDEF'";
+        qry += "            WHEN   40 THEN 'LVARCHAR'";
+        qry += "            WHEN   43 THEN 'LVARCHAR'";
+        qry += "            WHEN   45 THEN 'BOOLEAN'";
+        qry += "            WHEN   52 THEN 'BIGINT'";
+        qry += "            WHEN   53 THEN 'BIGSERIAL'";
+        qry += "            WHEN 2061 THEN 'IDSSECURITYLABEL'";
+        qry += "            WHEN 4118 THEN 'ROW'";
+        qry += "        END";
+        qry += "           AS data_type";
+        qry += "     , " + (forProcedureParameters ? "c.paramattr" : "1"); // Dummy value for table/view columns.
+        qry += "           AS direction";
+        qry += "     , c." + (forProcedureParameters ? "paramid" : "colno");
+        qry += "           AS position";
+        qry += "  FROM informix." + (forProcedureParameters ? "sysprocedures p" : "systables p");
+        qry += "     , informix." + (forProcedureParameters ? "sysproccolumns c" : "syscolumns c");
+        qry += " WHERE p." + (forProcedureParameters ? "procid" : "tabid") + " = c." + (forProcedureParameters ? "procid" : "tabid");
+        qry += "   AND ";
+
+        if (qualifiers.length == 2) {
+            qry += "LOWER(p.owner) = ? AND LOWER(p." + (forProcedureParameters ? "procname" : "tabname") + ") = ?";
+        } else {
+            qry += "LOWER(p." + (forProcedureParameters ? "procname" : "tabname") + ") = ?";
+        }
+
+        return qry;
+    }
+
+    public Map<String, DbParameterAccessor> getAllColumns(String tableOrViewName)
+            throws SQLException {
+        String[] qualifiers = NameNormaliser.normaliseName(tableOrViewName).split("\\.");
+        return readIntoParams(qualifiers, columnsOrParamtersQueryText(false, tableOrViewName));
+    }
+
+    private Map<String, DbParameterAccessor> readIntoParams(String[] queryParameters, String query) throws SQLException {
+        PreparedStatement dc = currentConnection.prepareStatement(query);
+        try {
```

Comment:
Similarly as the other discussion: might be cleaner to use `try-with-resources` syntax here too.